core/services/blockhashstore/feeder.go

```diff
@@ -63,6 +82,34 @@ type Feeder struct {
 	errsLock      sync.Mutex
 }
 
+func (f *Feeder) subscribeHeadBroadcaster() {
+	f.headBroadcaster.Subscribe(f)
+}
+
+// OnNewLongestChain implements HeadTrackable
+// note that the ctx has a timeout of 2 seconds
+// the below code cannot exceed that else we start missing
+// heartbeat bh's.
+func (f *Feeder) OnNewLongestChain(ctx context.Context, head *evmtypes.Head) {
```

Comment:
I don't think we need to use `OnNewLongestChain` and rely on the head broadcaster; I think we can just spawn a goroutine and use a waitgroup instead. The goroutine would listen to a timer and `<-ctx.Done()` and on every timer channel hit it would call `storeEarliest` in the BHS.

Reply:
In fact I think it would be preferable to avoid the head broadcaster altogether if possible.

Reply:
If we avoid the head broadcaster, do you prefer directly hitting RPC endpoints?

Reply:
Used Go timer and storeEarliest instead of headBroadcaster